make the prefix for environment variables alterable

Tim-Erwin opened the pull request:
Suppose my app is named "Abc", it is really not obvious, why the configuration prefix for environment variables is "SANIC_". Also, if I happen to have more than one Sanic app in one environment, there would be no way to distinguish those. Hence, I propose to make it possible to use other prefixes.

jrocketfingers commented:
LGTM, except that it'd be cool if we documented it too.

jrocketfingers commented:
https://github.com/channelcat/sanic/blob/master/docs/sanic/config.md#loading-configuration

This looks like a fine place to start.

Tim-Erwin commented:
Great, now I get to be the one to write the docs for something already existing *grumble*

Well, you're right, I'll add some docs :)

Tim-Erwin commented:
Oh, I just saw, there is already documentation for that, my fault. Of course I'll update it.

Tim-Erwin commented:
I just realized, that - while this change is sufficient for my use case - many people will probably use the `load_env` option. What do you think about allowing to pass in the prefix via this `load_env` variable? E.g.

```python
app = Sanic(load_env='MYAPP_')
```

Tim-Erwin commented:
I added the possibility to specify the prefix through the app constructor, added docs and tests.

Also I fixed the existing tests, two of the functions had the same name which caused one of them (the failing one) to be skipped.
